test/src/unittests/tonal/test_pitchcontours.py

```diff
@@ -0,0 +1,216 @@
+
+        #The reason is that 2D vectors dont reload easily from files, using the method shown below.
+        save('pitchcontourstarttimes_synthetic.npy',startTimes) 
+        save('pitchcontourbins_synthetic.npy', bins)               
+        save('pitchcontoursaliences_synthetic.npy', saliences)
+
+        # Captured from previous runs of pitch contours on the synthesis
+        expectedDuration = 0.00290249427780509
+        
+        loadedPitchContourBins = load(join(filedir(), 'pitchcontours/pitchcontourbins_synthetic.npy'))        
+        loadedPitchContourSaliences = load(join(filedir(), 'pitchcontours/pitchcontoursaliences_synthetic.npy'))
+        loadedPitchContourStartTimes = load(join(filedir(), 'pitchcontours/pitchcontourstarttimes_synthetic.npy'))
+
+        expectedPitchContourBins = loadedPitchContourBins.tolist() 
+        expectedPitchContourSaliences = loadedPitchContourSaliences.tolist() 
+        expectedPitchContourStartTimes = loadedPitchContourStartTimes.tolist() 
+
+        index = 0
+        while index<len(expectedPitchContourBins):
+           self.assertAlmostEqualVectorFixedPrecision(expectedPitchContourBins[index], bins[index], 8)
+           index+=1         
+
+        index = 0
+        while index<len(expectedPitchContourSaliences):
+           self.assertAlmostEqualVectorFixedPrecision(expectedPitchContourSaliences[index], saliences[index], 8)
+           index+=1         
+
+        self.assertAlmostEqualVectorFixedPrecision(expectedPitchContourStartTimes,startTimes, 8)        
+        self.assertAlmostEqual(expectedDuration,duration, 8)             
+
+ 
+    def testARealCase(self):
+        frameSize = 1024
+        sr = 44100
+        hopSize = 512
+        filename = join(testdata.audio_dir, 'recorded', 'vignesh.wav')
+        audio = MonoLoader(filename=filename, sampleRate=44100)()        
+
+        # Declare the algorithmns to be called (6 in total)
+        psf = PitchSalienceFunction()
+        psfp = PitchSalienceFunctionPeaks()
+        w = Windowing(type='hann', normalized=False)
+        spectrum = Spectrum()
+        spectralpeaks = SpectralPeaks()
+        pc = PitchContours()
+
+        # Populate an array of frame-wise vectors of cent bin values representing each contour
+        peakBins = []
+        # Populate a frame-wise array of values of salience function peaks
+        peakSaliences = []        
+        for frame in FrameGenerator(audio, frameSize=1024, hopSize=hopSize,
+                                    startFromZero=True):
+            freq_peaks, mag_peaks = spectralpeaks(spectrum(w(frame)))
+            len_freq_peaks = len(freq_peaks)
+            len_mag_peaks = len(mag_peaks)
+            
+            # Skip the first vector elements that correspond to zero frequency
+            freq_peaksp1 = freq_peaks[1:len_freq_peaks]
+            # Do the same for mag. vector to keep the lengths the same
+            mag_peaksp1 = mag_peaks[1:len_mag_peaks]
```

Comment:
We don't need to skip those, because we expect the PitchSalienceFunction to ignore peaks below its reference frequency.